### DISCUSSION TITLE
Quasar Config TypeScript & ESM

### DISCUSSION THREAD

**Discussion by angelhdzmultimedia** [quasarframework/quasar#15286] [Ideas / Proposals] [answered]
1. Any plans to replace Quasar's config to quasar.config.ts and ESM?
2. If there are no plans, or if the plans are far in the future, is there a workaround?

Thank you.

**Comment by rstoenescu** [comment, marked as answer]
Yes. The global CLI has already been ported to ESM. I'm currently working on porting the local CLIs (q/app-webpack & q/app-vite). This will include support for quasar.config file in ESM and TS. And SSR on ESM format out of the box. And for q/app-vite it will also mean the upgrade to Vite 4. Once I have everything figured out I will make an announcement about it.

**Reply by angelhdzmultimedia** [reply]
> Yes. The global CLI has already been ported to ESM. I'm currently working on porting the local CLIs (q/app-webpack & q/app-vite). This will include support for quasar.config file in ESM and TS. And SSR on ESM format out of the box. And for q/app-vite it will also mean the upgrade to Vite 4. Once I have everything figured out I will make an announcement about it.

Beautiful! Wishing you health and success, and I appreciate all you have done. Congrats for an awesome UI framework for Vue 3.

**Reply by ScreamZ** [reply]
Is it live ?

**Reply by rstoenescu** [reply]
Yes. Check the Announcements.
